lib/oxide-vpc/src/engine/overlay.rs

```diff
@@ -379,10 +379,16 @@ impl StaticAction for DecapAction {
     ) -> GenHtResult {
         match &pkt_meta.outer.encap {
             Some(EncapMeta::Geneve(geneve)) => {
-                action_meta.insert(
-                    ACTION_META_VNI.to_string(),
-                    geneve.vni.to_string(),
-                );
+                // XXX: This is now slightly overloaded to mean "packet
+                //      originated from the VPC which has this VNI".
+                //      Unsure if we want to have sled agent add a
+                //      VNI && (not external) predicate.
```

Comment:
I think that's fine, the user facing firewall API doesn't even deal in VNI's directly, but rather VPC originated traffic and sled-agent translates it to the corresponding VNI.

If we did want to separate out things a bit, perhaps instead we could add a new meta tag to indicate internal/external and then update the [predicate generated for `Address::Vni`](https://github.com/oxidecomputer/opte/blob/98d33125413f01722947e322f82caf9d22209434/lib/oxide-vpc/src/engine/firewall.rs#L211-L214):

```rust
            (_, Address::Vni(vni)) => Some(Predicate::And(
                Predicate::Meta(ACTION_META_EXTERNAL.to_string(), "false".to_string()),
                Predicate::Meta(ACTION_META_VNI.to_string(), vni.to_string()),
            )),
```
(`Predicate::And` doesn't exist but i imagine would be straight forward)

Otherwise, I would at the very least add a comment here explaining why we only conditionally add `ACTION_META_VNI`.

Reply:
Thanks, Luqman -- I'll explain this via comment for now, though I may have one or two tweaks after testing with sidecar-lite!